services/core/java/com/android/server/clipboard/ClipboardService.java

```diff
@@ -1615,4 +1647,9 @@ private TextClassificationManager createTextClassificationManagerAsUser(@UserIdI
         Context context = getContext().createContextAsUser(UserHandle.of(userId), /* flags= */ 0);
         return context.getSystemService(TextClassificationManager.class);
     }
+
+    private boolean isReadBlockedForPkg(int uid, String pkg, int userId, Clipboard clipboard) {
+        return !UserHandle.isSameApp(uid, clipboard.primaryClipUid)
```

Comment:
isSameApp check implies that app should have access to its own clipboard contents across user profiles, which is blocked elsewhere